### PR TITLE
Stricter formatParams

### PR DESCRIPTION
**Pull request by lukaw3d**

### CODE REVIEW

File: src/locales/__tests__/i18n-formatting.test.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ test('formatting dates with i18n', () => {
           timeZoneName: 'short',
           timeZone: 'UTC',
           locale: 'en-US',
-        },
+        } satisfies Intl.DateTimeFormatOptions & { locale?: Intl.BCP47LanguageTag },
```
Comment (csillag) [inline src/locales/__tests__/i18n-formatting.test.ts:22]: Shouldn't we have a name for this union?

**Review by csillag — APPROVED** [review]
Reviewed. LGTM.

**Review by buberdds — APPROVED** [review]

File: src/locales/__tests__/i18n-formatting.test.ts (hunk 1 above)
Comment (lukaw3d) [reply src/locales/__tests__/i18n-formatting.test.ts:22]: I think that's needlessly complex just for tests. Unfortunately I'm not able to make a strict type with 
```ts
      (Intl.DateTimeFormatOptions & { locale?: Intl.BCP47LanguageTag })
    | (Intl.RelativeTimeFormatOptions & { locale?: Intl.BCP47LanguageTag })
    | (Intl.NumberFormatOptions & { locale?: Intl.BCP47LanguageTag })
    | (Intl.ListFormatOptions & { locale?: Intl.BCP47LanguageTag })
```

Comment (lukaw3d) [reply src/locales/__tests__/i18n-formatting.test.ts]: or i could just remove locale  https://github.com/oasisprotocol/explorer/pull/115/commits/3f951c4d38e17d87e3c1b1b8f64b73613256cc99

**Pull request merged**